lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -705,14 +713,28 @@ class ExprContextAnalyzer {
     }
   }
 
+  static bool isSingleExpressionBodyForCodeCompletion(BraceStmt *body) {
+    switch (body->getNumElements()) {
+    case 0:
+      // Nothing in the body except the code-completion token.
+      return true;
+    case 1:
+      return body->getElements()[0].is<Expr *>();
+    default:
+      return false;
```

Comment:
Do we already have test cases for this path? like:
```swift
var floatVal: Float = 2
let _: () -> Float = {
  #^COMPLETE^# // Should be normal completion.
  return 42
}
let _: () -> SomeEnum = {
  .#^COMPLETE^# // Should not provide anything.
  return 42
}
```
If not, please add them.

Reply:
Good catch, this code for handling `getNumElements() == 0` is from before I changed the parser to add the `CodeCompletionExpr` in `completeStmtOrExpr`.  I'll remove the 0 case and add the additional test.